TGEngine/io/Resource.cpp

```diff
@@ -20,46 +20,44 @@ void loadResourceFile(const char* name, Map* map) {
 
 	uint32_t blocklength = 0;
 
-	map->textures.reserve(2048);
+	std::vector<TextureInputInfo> textureBindingInfos;
+	textureBindingInfos.reserve(2048);
 
 	fread(&blocklength, sizeof(uint32_t), 1, file);
 
 	while (blocklength != UINT32_MAX)
 	{
 		stbi_uc* resbuffer = new stbi_uc[blocklength];
-		uint32_t test = fread(resbuffer, sizeof(stbi_uc), blocklength, file);
+		fread(resbuffer, sizeof(stbi_uc), blocklength, file);
 
 		TextureInputInfo inputInfo;
 		inputInfo.data = stbi_load_from_memory(resbuffer, (int)blocklength, &inputInfo.x, &inputInfo.y, &inputInfo.comp, STBI_rgb_alpha);
-		map->textures.push_back(inputInfo);
+		textureBindingInfos.push_back(inputInfo);
 
 		fread(&blocklength, sizeof(uint32_t), 1, file);
 	}
 
-	if (feof(file))
-		return;
-
 	blocklength = 0;
 
-	map->materials.reserve(256);
+	createdMaterials = new Material[256];
 
 	fread(&blocklength, sizeof(uint32_t), 1, file);
 
+	uint32_t materialID = 0;
+
 	while (blocklength != UINT32_MAX)
 	{
 		Material material;
 		fread(&material, sizeof(Material), 1, file);
 
-		map->materials.push_back(material);
+		createdMaterials[materialID++] = material;
 
 		fread(&blocklength, sizeof(uint32_t), 1, file);
 	}
 
-	if (feof(file))
-		return;
-
 	// Start to read the actor
-	map->actors.reserve(2048);
+	std::vector<ActorInputInfo> actorInputInfos;
+	actorInputInfos.reserve(2048);
```

Comment:
magic number? what does it mean?

Reply:
This is a SPECULATIVE reserve, means: This could be any number, it does not matter for the program logic at all. In this case tho it makes sense to use the maximum number of textures allowed by the current shader. HOWEVER the maximum number could change and it would only make a difference in performance.